File: src/cli/Clients/ProcessorClient.cs
using Arma.Demo.Core.Processing;
using Arma.Demo.Core.Sync;

namespace CloudCli.Clients;
public class ProcessorClient : SyncService<Package>
{
    public ProcessorClient(string endpoint) : base(endpoint)
    {
        OnPush = OnNotify = Output;
    }

    static Task Output(SyncMessage<Package> message)
    {
        Console.WriteLine(message.Message);
        return Task.CompletedTask;
    }
}

File: tests/cross-service-signalr/app/spa/src/app/services/processor-client.ts
import { Injectable } from '@angular/core';
import { SyncClient } from './sync-client';

import {
    Package,
    SyncActionType,
    SyncMessage
} from '../models';

@Injectable({
    providedIn: 'root'
})
export class ProcessorClient extends SyncClient<Package> {
    private initEvents = () => {
        this.onPush = this.output;
        this.onNotify = this.output;
        this.onComplete = this.output;
        this.onReturn = this.output;
        this.onReject = this.output;
    }

    constructor() {
        super('http://localhost:5100/processor');
        this.initEvents();
    }

    output = (message: SyncMessage<Package>) => {
        console.log(`${SyncActionType[message.action]}: {message.message}`);
        this.messages.push(`${SyncActionType[message.action]}: {message.message}`);
    }
}